### PR TITLE
Schedule rruleset fix related #13446

### PR DESCRIPTION
**Pull request by KaraokeKev**
##### SUMMARY

Fix to process_list function. isinstance checks for list, but list is desired end state. Changed to string. Related #13446

In addition, added .lower() to value definition to allow for case-insensitivity from user input.

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - Collection

##### AWX VERSION

```
awx: 0.1.dev32853+g37c7a0c
```

##### ADDITIONAL INFORMATION

**Case 1**
```
rule_parameters:
        - frequency: 'day'
          interval: 1
        - frequency: 'day'
          interval: 1
          byweekday: sunday
          include: False
```
Prior to change:
```
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'awx.awx.schedule_rruleset'. Error was a <class 'ansible.errors.AnsibleError'>, original message: In rule 2 byweekday must only contain values in monday, tuesday, wednesday, thursday, friday, saturday, sunday. In rule 2 byweekday must only contain values in monday, tuesday, wednesday, thursday, friday, saturday, sunday"}
```
Following change:
```
ok: [localhost] => {
    "msg": "DTSTART;TZID=America/New_York:20220919T042000 RRULE:FREQ=DAILY;INTERVAL=1 EXRULE:FREQ=DAILY;BYDAY=SU;INTERVAL=1"
}
```

**Case 2**
```
rule_parameters:
        - frequency: 'day'
          interval: 1
        - frequency: 'day'
          interval: 1
          byweekday: sunday,tuesday
          include: False
```
Prior to change:
```
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'awx.awx.schedule_rruleset'. Error was a <class 'ansible.errors.AnsibleError'>, original message: In rule 2 byweekday must only contain values in monday, tuesday, wednesday, thursday, friday, saturday, sunday. In rule 2 byweekday must only contain values in monday, tuesday, wednesday, thursday, friday, saturday, sunday"}
```
Following change:
```
ok: [localhost] => {
    "msg": "DTSTART;TZID=America/New_York:20220919T042000 RRULE:FREQ=DAILY;INTERVAL=1 EXRULE:FREQ=DAILY;BYDAY=TU,SU;INTERVAL=1"
}
```

**Case 3**
```
rule_parameters:
        - frequency: 'day'
          interval: 1
        - frequency: 'day'
          interval: 1
          byweekday: ['sunday', 'tuesday']
          include: False
```
Prior to change:
```
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'awx.awx.schedule_rruleset'. Error was a <class 'AttributeError'>, original message: 'list' object has no attribute 'split'. 'list' object has no attribute 'split'"}
```
Following change:
```
ok: [localhost] => {
    "msg": "DTSTART;TZID=America/New_York:20220919T042000 RRULE:FREQ=DAILY;INTERVAL=1 EXRULE:FREQ=DAILY;BYDAY=TU,SU;INTERVAL=1"
}
```

**Case 4**
```
rule_parameters:
        - frequency: 'day'
          interval: 1
        - frequency: 'day'
          interval: 1
          byweekday: sunday, monday, Tuesday
          include: False
```
Prior to change:
```
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'awx.awx.schedule_rruleset'. Error was a <class 'ansible.errors.AnsibleError'>, original message: In rule 2 byweekday must only contain values in monday, tuesday, wednesday, thursday, friday, saturday, sunday. In rule 2 byweekday must only contain values in monday, tuesday, wednesday, thursday, friday, saturday, sunday"}
```
Following change:
```
ok: [localhost] => {
    "msg": "DTSTART;TZID=America/New_York:20220919T042000 RRULE:FREQ=DAILY;INTERVAL=1 EXRULE:FREQ=DAILY;BYDAY=MO,TU,SU;INTERVAL=1"
}
```

### CODE REVIEW

**Comment by fosterseth** [comment]
@KaraokeKev thanks for this PR, do you mind adding a couple of test cases to this file (to assert that the lookup is successful)

https://github.com/ansible/awx/blob/devel/awx_collection/tests/integration/targets/lookup_rruleset/tasks/main.yml

**Comment by fosterseth** [comment]
duplicate PR (that aims to fix the same issue) is here https://github.com/ansible/awx/pull/13541

We plan to merge that in, so this PR may no longer be needed.

**Comment by KaraokeKev** [comment]
> 

I would note that while #13541 does solve the base bug, it does not allow for the passing of a list or case insensitivity.

**Comment by AlanCoding** [comment]
If we could get a test case added to `awx_collection/tests/integration/targets/lookup_rruleset/tasks/main.yml` which demonstrates the case insensitivity bug then that should allow us to tease apart the unique fixes here eventually.

**Comment by KaraokeKev** [comment]
> If we could get a test case added to `awx_collection/tests/integration/targets/lookup_rruleset/tasks/main.yml` which demonstrates the case insensitivity bug then that should allow us to tease apart the unique fixes here eventually.

Added.

File: awx_collection/plugins/lookup/schedule_rruleset.py [hunk 1]
```diff
@@ -210,10 +210,10 @@ def process_integer(self, field_name, rule, min_value, max_value, rule_number):
 
     def process_list(self, field_name, rule, valid_list, rule_number):
         return_values = []
-        if isinstance(rule[field_name], list):
+        if isinstance(rule[field_name], str):
```
Comment (AlanCoding) [inline awx_collection/plugins/lookup/schedule_rruleset.py]: I don't agree with this. If you look elsewhere in this file you find that this line is duplicated, and the other case has an associated comment, see https://github.com/ansible/awx/pull/13541#issuecomment-1442134477 for my approach to address the original intent. Best to abandon this here.

File: awx_collection/plugins/lookup/schedule_rruleset.py [hunk 2]
```diff
@@ -210,10 +210,10 @@ def process_integer(self, field_name, rule, min_value, max_value, rule_number):
...
             rule[field_name] = rule[field_name].split(',')
         for value in rule[field_name]:
-            value = value.strip()
+            value = value.strip().lower()
```
Comment (AlanCoding) [inline awx_collection/plugins/lookup/schedule_rruleset.py:217]: Agree. Thanks for this fix and associated test. Hopefully I can get the other patch merged, then this rebased on top of it.

**Comment by AlanCoding** [comment]
This has conflicts due to https://github.com/ansible/awx/pull/13541, but I hope the issues are easy to follow from my review comments here. I would love to get this rebased and then merged.

**Comment by rebeccahhh** [comment]
@KaraokeKev would you like to do the rebase? If not a member of our team can try to get around to this at some point in time (it will probably be merged faster if you do it though).

**Comment by KaraokeKev** [comment]
> @KaraokeKev would you like to do the rebase? If not a member of our team can try to get around to this at some point in time (it will probably be merged faster if you do it though).

Rebase completed

File: awx_collection/tests/integration/targets/lookup_rruleset/tasks/main.yml [hunk 3]
```diff
@@ -356,3 +356,20 @@
...
+    rrules:
+      - frequency: 'day'
+        interval: 1
+        byweekday: monday, Tuesday, WEDNESDAY
+
+- assert:
+    that:
+      - results is success
+      - "'DTSTART;TZID=UTC:20220430T103045 RRULE:FREQ=DAILY;BYDAY=MO,TU,WE;INTERVAL=1' == complex_rule"
+
```
Comment (djyasin) [inline awx_collection/tests/integration/targets/lookup_rruleset/tasks/main.yml]: @KaraokeKev Hello! It looks like the blank line on 375 is causing our api-linting check to fail. `375:1 error too many blank lines (1 > 0) (empty-lines)`

I believe removing this line should correct the error and get that check to a passing state.

**Review by djyasin — APPROVED** [review]

**Review by jay-steurer — APPROVED** [review]

**Pull request merged**